### PR TITLE
feat: (IAC-644) Add DaC support for crunchy v5 

### PR DESCRIPTION
**Pull request by dhoucgitter**
# Changes:
- DaC changes related to supporting internal v5 crunchy postgres db.
- Crunchy v5 support for CDS Postgres is still to be added.
- (IAC-724) Update max_connections for Crunchy v5 internal postgres to support Multi-tenancy
- DaC changes related to supporting external crunchy postgres DBs

# Tests:
See internal tickets for details

### CODE REVIEW

File: roles/vdm/tasks/postgres/postgres.yaml [hunk 1]
```diff
@@ -111,7 +110,7 @@
         cadence_number: "{{ V4_CFG_CADENCE_VERSION }}"
         existing: "{{ vdm_overlays }}"
         add:
-          - { transformers: "overlays/external-postgres/external-postgres-transformer.yaml" }
+          - { resources: "overlays/postgres/platform-postgres", min: "2022.10" }
```
Comment (thpang) [inline roles/vdm/tasks/postgres/postgres.yaml:113]: What if I have an older order, what and how does the external-postgres-transformer.yaml get put in place as the v5 operator will not be in the mix?

**Review by thpang — COMMENTED** [review]
Need clarification on code adjustment.

File: roles/vdm/tasks/mirror.yaml [hunk 2]
```diff
@@ -19,13 +19,30 @@
...
-      - { generators: "mirror.yaml", vdm: true }
+  block:
+    - name: v4 crunchy
+      overlay_facts:
+        cadence_name: "{{ V4_CFG_CADENCE_NAME }}"
+        cadence_number: "{{ V4_CFG_CADENCE_VERSION }}"
+        existing: "{{ vdm_overlays }}"
+        add:
+          - { transformers: "mirror.yaml", vdm: true, priority: 65 }
+          - { generators: "mirror.yaml", vdm: true }
```
Comment (thpang) [inline roles/vdm/tasks/mirror.yaml:30]: Replace these:
```yaml
- { transformers: "mirror.yaml", vdm: true, priority: 65 }
- { generators: "mirror.yaml", vdm: true }
```
with:
```yaml
- { transformers: "mirror.yaml", vdm: true, priority: 65, max: 2022.09 }
- { generators: "mirror.yaml", vdm: true, max: 2022.09 }
```
Then remove the `when` statement below.

File: roles/vdm/tasks/mirror.yaml [hunk 3]
```diff
@@ -19,13 +19,30 @@
...
+        - V4_CFG_CADENCE_VERSION is version('2022.10', "<")
+        - not V4_CFG_CADENCE_NAME|lower == "fast"
+    - name: v5 crunchy
+      overlay_facts:
+        cadence_name: "{{ V4_CFG_CADENCE_NAME }}"
+        cadence_number: "{{ V4_CFG_CADENCE_VERSION }}"
+        existing: "{{ vdm_overlays }}"
+        add:
+          - { transformers: "mirror.yaml", vdm: true, priority: 65 }
+          - { generators: "mirror_v2.yaml", vdm: true }
```
Comment (thpang) [inline roles/vdm/tasks/mirror.yaml:41]: Replace these lines:
```yaml
          - { transformers: "mirror.yaml", vdm: true, priority: 65 }
          - { generators: "mirror_v2.yaml", vdm: true }
```
with:
```yaml
          - { transformers: "mirror.yaml", vdm: true, priority: 65, max: 2022.09 }
          - { generators: "mirror.v2.yaml", vdm: true, min: 2022.10 }
```
Also remove the `when` below. And change the versioned file names from _v2 -> .v2.

**Review by thpang — CHANGES_REQUESTED** [review]
Look over logic.

File: roles/vdm/tasks/mirror.yaml (hunk 2 above)
Comment (dhoucgitter) [reply roles/vdm/tasks/mirror.yaml]: Done.

File: roles/vdm/tasks/mirror.yaml (hunk 3 above)
Comment (dhoucgitter) [reply roles/vdm/tasks/mirror.yaml]: Done.

**Review by sayeun — APPROVED** [review]
Looks okay to me.  I did ran our test automation to deploy an older cadence to make sure it still work.

**Review by thpang — CHANGES_REQUESTED** [review]

**Review by thpang — APPROVED** [review]
LGTM. Do we need a code review of this one. Seems like a lot of extra items for v5 vs v4 and just want to be clear what was deprecated and what's new.

File: roles/vdm/templates/generators/mirror_v2.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
```
[the remark this answers is not in the document]
Comment (dhoucgitter) [reply roles/vdm/templates/generators/mirror_v2.yaml]: The file has been removed.

**Review by sayeun — APPROVED** [review]
Based on the latest test results in IAC-644 and my successful 2022.1.4 deployment (kustomization.yaml has 2 mirror entries), I approve this.

**Pull request merged**